lib/usd/ui/layerEditor/layerTreeView.cpp

```diff
@@ -131,7 +136,77 @@ void LayerTreeView::onItemDoubleClicked(const QModelIndex& index)
     }
 }
 
-void LayerTreeView::onModelReset() { expandAll(); }
+LayerViewMemento::LayerViewMemento(const LayerTreeView& view, const LayerTreeModel& model)
+{
+    preserve(view, model);
+}
+
+void LayerViewMemento::preserve(const LayerTreeView& view, const LayerTreeModel& model)
+{
+    const LayerItemVector items = model.getAllItems();
+    if (items.size() == 0)
+        return;
+
+    for (const LayerTreeItem* item : items) {
+        if (!item)
+            continue;
+
+        PXR_NS::SdfLayerRefPtr layer = item->layer();
+        if (!layer)
+            continue;
+
+        const ItemId    id = layer->GetIdentifier();
+        const ItemState state = { view.isExpanded(item->index()) };
+
+        itemsState[id] = state;
+    }
+}
+
+void LayerViewMemento::restore(LayerTreeView& view, LayerTreeModel& model)
+{
+    const LayerItemVector items = model.getAllItems();
+
+    for (const LayerTreeItem* item : items) {
+        if (!item)
+            continue;
+
+        bool expanded = true;
+
+        PXR_NS::SdfLayerRefPtr layer = item->layer();
+        if (layer) {
+            const ItemId id = layer->GetIdentifier();
+            const auto   state = itemsState.find(id);
+            if (state != itemsState.end()) {
+                expanded = state->second.expanded;
+            }
+        }
+
+        view.setExpanded(item->index(), expanded);
```

Comment:
When you call things like this you trigger a redraw. So normally when doing this in a loop, we'll first disable drawing updates before the loop, and then enable them (and redraw) after the loop.

Reply:
Do we have a utility class that disable Qt redraws and trigger and repaint already existing?